### PR TITLE
ghc: 8.10.6

### PR DESCRIPTION
**Pull request by fishtreesugar**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
We can update the bootstrap resources too. I think we should be able to use 8.10.6 on Intel now.

**Comment by carlocab** [comment]
```
Linking utils/ghc-cabal/dist/build/tmp/ghc-cabal ...
Undefined symbols for architecture x86_64:
  "___darwin_check_fd_set_overflow", referenced from:
      _awaitEvent in libHSrts.a(Select.o)
ld: symbol(s) not found for architecture x86_64
```
Looks like this hasn't been fixed yet. I think we need to keep bootstrapping with 8.10.4 on Intel.

File: Formula/ghc.rb [hunk 1]
```diff
@@ -43,42 +42,34 @@ class Ghc < Formula
...
+          url "https://downloads.haskell.org/~ghc/8.10.4/ghc-8.10.4-x86_64-apple-darwin.tar.xz"
+          sha256 "725ecf6543e63b81a3581fb8c97afd21a08ae11bc0fa4f8ee25d45f0362ef6d5"
+        end
       else
-        url "https://downloads.haskell.org/ghc/8.10.5/ghc-8.10.5-aarch64-apple-darwin.tar.xz"
-        sha256 "03684e70ff03d041b9a4e0f84c177953a241ab8ec7a028c72fa21ac67e66cb09"
+        url "https://downloads.haskell.org/ghc/8.10.6/ghc-8.10.6-aarch64-apple-darwin.tar.xz"
+        sha256 "9e43fc3a39d2f2762262c63868653984e381e29eff6386f7325aad501b9190ad"
       end
     end
```
Comment (carlocab) [inline Formula/ghc.rb:64]: You should probably do something like
```ruby
if Hardware::CPU.arm? || MacOS.version > :mojave
  ...
else
  ...
end
```

**Comment by fishtreesugar** [comment]
@mpickering are we still suffering the issue fixed in https://gitlab.haskell.org/ghc/ghc/-/merge_requests/5992? 

build log:
https://github.com/Homebrew/homebrew-core/runs/3329401373#step:7:3463

**Comment by mpickering** [comment]
What exactly are you trying to build with what? It seems an issue with the make build system we are not normally using.

**Comment by fishtreesugar** [comment]
@mpickering  In the build log I pasted, we trying to bootstrap GHC 8.10.6 w/  binary GHC 8.10.6 in MacOS 10.14 (mojave)

**Comment by mpickering** [comment]
Sorry, yes I can reproduce this locally..

**Comment by fishtreesugar** [comment]
@carlocab test failed with weird error message, anything we can do let it move forward?

**Comment by SMillerDev** [comment]
`allureofthestars` `haskel-language-server` and `agda` need to have their revision bumped.

**Comment by carlocab** [comment]
Please modify only a single formula per commit. (That's where the `automerge-skip` label came from.) Fixed that for you. When bumping revisions, you can do `brew bump-revision` to make sure it's done in the right format.

**Comment by fishtreesugar** [comment]
note: https://gitlab.haskell.org/ghc/ghc/-/merge_requests/6360 upstream fix of `___darwin_check_fd_set_overflow` issue

**Comment by carlocab** [comment]
`stellar-core` depends on `ghc` only via `pandoc` -- which is presumably used only to build docs. I doubt it is related to this version bump.

We should try to figure out the `cedille` failure though.

**Comment by fishtreesugar** [comment]
I am trying to fix `cedille` failure in #83560

**Review by carlocab — DISMISSED** [review]
Thanks @fishtreesugar!

Looking at it more closely, it looks like `cedille` is pre-existing. It was also seen at #80549.

**Comment by carlocab** [comment]
Oh, there is a merge conflict for `allureofthestars`. I'll just merge this manually to avoid having to re-run CI.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**